File: Week2/김다빈/BOJ_2422_3/2422.py
n, m = map(int, input().split())
iceCreams = [i for i in range(1, (n + 1))]
forbiddenArr = [[False for _ in range(n + 1)] for _ in range(n + 1)]
answer = 0

for _ in range(m):
    ice1, ice2 = map(int, input().split())
    forbiddenArr[ice1][ice2] = True
    forbiddenArr[ice2][ice1] = True

for i in range(1, n + 1):
    for j in range(i + 1, n + 1):
        for k in range(j + 1, n + 1):
            if not forbiddenArr[i][j] and not forbiddenArr[i][k] and not forbiddenArr[j][k]:
                answer += 1

print(answer)

File: Week2/김다빈/BOJ_2422_3/2422.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int cnt = 0;
        boolean[][] forbiddenArr = new boolean[n + 1][n + 1];

        for(int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int ice1 = Integer.parseInt(st.nextToken());
            int ice2 = Integer.parseInt(st.nextToken());
            forbiddenArr[ice1][ice2] = true;
            forbiddenArr[ice2][ice1] = true;
        }


        for (int i = 1; i <= n; i++) {
            for (int j = i + 1; j <= n; j++) {
                for (int k = j + 1; k <= n; k++) {
                    if (!forbiddenArr[i][j] && !forbiddenArr[j][k] && !forbiddenArr[i][k]) {
                        cnt++;
                    }
                }
            }
        }

        System.out.println(cnt);
    }
}